libs/core/type_support/tests/unit/CMakeLists.txt

```diff
@@ -8,6 +8,9 @@ set(tests relocate relocate_at uninitialized_relocate_n_primitive)
 
 if(HPX_WITH_CXX20_COROUTINES)
   set(tests ${tests} generator)
+  if(HPX_WITH_CXX23_STD_GENERATOR)
+    set(generator_FLAGS ${HPX_CXX23_FLAG})
+  endif()
```

Comment:
What's the rationale of this change? The idea of `HPX_WITH_CXX23_STD_GENERATOR` is to be enabled if std::generator is available, for some platforms that may be true even if compiled in C++20 mode? Also, where is `HPX_CXX23_FLAG` defined?